pkg/client/client.go

```diff
@@ -109,20 +66,41 @@ func newClient(host string) *HTTPClient {
 		IdleConnTimeout:       90 * time.Second,
 		TLSHandshakeTimeout:   5 * time.Second,
 		ExpectContinueTimeout: 1 * time.Second,
+		DisableKeepAlives:     false,
 	}
-	transport.DisableKeepAlives = false
 	maxConnPerHost := viper.GetInt(optname.MaxConnPerHost)
 	if maxConnPerHost > 0 {
-		transport.MaxConnsPerHost = maxConnPerHost
+		baseTransport.MaxConnsPerHost = maxConnPerHost
 	}
 
-	client := &http.Client{
-		Transport:     &R8GetRetryingRoundTripper{Transport: transport},
-		CheckRedirect: checkRedirectFunc,
+	transport := &UserAgentTransport{Transport: &baseTransport}
+
+	retryClient := &retryablehttp.Client{
+		HTTPClient: &http.Client{
+			Transport:     transport,
+			CheckRedirect: checkRedirectFunc,
+		},
+		Logger:       nil,
+		RetryWaitMin: retryMinWait,
+		RetryWaitMax: retryMaxWait,
+		RetryMax:     viper.GetInt(optname.Retries),
+		CheckRetry:   retryablehttp.DefaultRetryPolicy,
+		Backoff:      backoffFunc,
 	}
+
+	client := retryClient.StandardClient()
 	return &HTTPClient{Client: client, host: host}
 }
 
+// backoffFunc is a wrapper around retryablehttp.DefaultBackoff that allows for adding a random jitter to the backoff
+// we utilize the jitter to avoid thundering herd issues since we are running with significant numbers of concurrent
+// downloads.
+func backoffFunc(min, max time.Duration, attemptNum int, resp *http.Response) time.Duration {
```

Comment:
Did you see https://pkg.go.dev/github.com/hashicorp/go-retryablehttp ? For some reason upstream provides strict exponential or jittered linear backoff so I can see how you might have got here anyway.

Reply:
I think we can just go with the linear jitter. I'll respin this PR to make that change as it lightens our carried code.

Reply:
I wanted the Retry-After header support so here is a version that covers LinearBackoffWithJitter *and* Retry-After header.